cql/src/main/java/org/apache/cassandra/stargate/transport/internal/messages/AuthResponse.java

```diff
@@ -72,7 +72,11 @@ public AuthResponse(byte[] token) {
       byte[] challenge = negotiator.evaluateResponse(token);
       if (negotiator.isComplete()) {
         AuthenticatedUser authenticatedUser = negotiator.getAuthenticatedUser();
-        persistenceConnection().login(authenticatedUser);
+        if (!(authenticatedUser.usesTransitionalAuth()
```

Comment:
Why do we have such logic?
If the user is not `usesTransitionalAuth` and `cql_use_transitional_auth` set to `true`?
It looks a bit convoluted to me. Could you elaborate more?

Reply:
`usesTransitionalAuth` will be set to true in the event that authentication is happening with something other than the traditional user/pass found in system_auth.roles but someone still might want to rely on C* permissions for authz. In the event that you're transitioning from purely C* authz to another system you can set `cql_use_transitional_auth` to true to say "if the authentication was with a different system AND transitional auth is supported then skip authenticating against C*"

Reply:
I think what's confusing is that we use very similar names for two different things (see also [Dmitri's comment](https://github.com/stargate/stargate/pull/502/files#r538533657) above).

Maybe renaming the Java method to something like `authenticatedUser.isFromExternalAuth()` would help.